code/menuui/techmenu.cpp

```diff
@@ -235,14 +236,10 @@ typedef struct {
 	int textures_loaded;	// if the model has textures loaded for it or not (hacky mem management)
 } tech_list_entry;
 
-static tech_list_entry *Ship_list = nullptr;
-static int Ship_list_size = 0;
-static tech_list_entry *Weapon_list = nullptr;
-static int Weapon_list_size = 0;
-static tech_list_entry *Intel_list = nullptr;
-static int Intel_list_size = 0;
-static tech_list_entry *Current_list;								// points to currently valid display list
-static int Current_list_size = 0;
+SCP_vector<tech_list_entry> Ship_list;
+SCP_vector<tech_list_entry> Weapon_list;
+SCP_vector<tech_list_entry> Intel_list;
+SCP_vector<tech_list_entry> Current_list;								// A copy of the currently valid display list
```

Comment:
These should be `static`. If something is defined globally, but not needed in other files, it should always be declared `static`.

Reply:
All right, I'll try to remember this for the future.  Sorry us newer c++ programmers have a lot that we're not yet used to.

Reply:
Well, that's why we have code reviews.  Not every person knows about every code technique.

Reply:
Most people don't seem to do it. Older C guys or embedded coders are the main users of those optimizations I think. I learned it from the far more experienced coders that worked on the initial icculus.org freespace2 port. Now it's just one of those easy things that I try to pass on to others :)